src/yunohost/data_migrations/0020_ssh_sftp_permissions.py

```diff
@@ -0,0 +1,50 @@
+import subprocess
+
+from moulinette import m18n
+from moulinette.utils.log import getActionLogger
+from moulinette.utils.filesystem import read_yaml
+
+from yunohost.tools import Migration
+from yunohost.permission import user_permission_update
+
+logger = getActionLogger('yunohost.migration')
+
+###################################################
+# Tools used also for restoration
+###################################################
+
+
+class MyMigration(Migration):
+    """
+        Add new permissions around SSH/SFTP features
+    """
+
+    dependencies = ["extend_permissions_features"]
+
+    @ldap_migration
+    def run(self, *args):
+
+        from yunohost.utils.ldap import _get_ldap_interface
+        ldap = _get_ldap_interface()
+
+        # Add SSH and SFTP permissions
+        ldap_map = read_yaml('/usr/share/yunohost/yunohost-config/moulinette/ldap_scheme.yml')
+
+        ldap.add("cn=ssh.main,ou=permission", ldap_map['depends_children']["cn=ssh.main,ou=permission"])
+        ldap.add("cn=sftp.main,ou=permission", ldap_map['depends_children']["cn=sftp.main,ou=permission"])
+
+        # Add a bash terminal to each users
+        users = ldap.search('ou=users,dc=yunohost,dc=org', filter="(loginShell=*)", attrs=["dn", "uid", "loginShell"])
+        for user in users:
+            if user['loginShell'][0] == '/bin/false':
+                dn=user['dn'][0].replace(',dc=yunohost,dc=org', '')
+                ldap.update(dn, {'loginShell': ['/bin/bash']})
+            else:
+                user_permission_update("ssh.main", add=user["uid"][0], sync_perm=False)
+
+        permission_sync_to_user()
```

Comment:
So I'm once more reviewing this and this ends up being catastrophic in the not-so-uncommon case where people manually edited their ssh conf.

In such a case, the SSH restriction on root+admin+members of ssh.main won't be applied, yet we're setting /bin/bash as the shell for all users. Which ends up with every users obtaining ssh access, which is not acceptable.

On the other hand, we can't just force-update the ssh configuration, because that may simply break the SSH workflow etc...

And ideally we don't want to just not change the loginShell, because /bin/false is legacy stuff and we won't have any way to reset the loginShell ...

Reply:
Follow-up of this discussion,

considering that we addressed a large bunch of the issue of "users can browse /var/www/$app" with the recent security fix, and after discussing with ljf, we propose to handle this situation like this: 
- keep applying what the current migration does
- the sshd_config won't be updated because the user manually edited it
- hence every user end up with an actual SSH access, which is "not cool" but not that bad because they cant browse /var/www/ or other critical locations
- we shall add a check + error message in the diagnosis if the sshd_config contains no `AllowGroups` or `AllowUsers` instructions to incentivize the admin to fix its config
- we can also perform this check and display this error during the migration